vars/githubWorkflowRun.groovy

```diff
@@ -0,0 +1,88 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+
+  This step triggers workflow run on Github actions.
+
+  Example:
+     script {
+       def args = [
+           repo: "owner/repository",
+           workflow: "build.yml", 
+           ref: "main",
+           parameters: [
+             path: "filebeat"
+           ],
+           credentialsId: "github-workflow-token",
+           ghVersion: "2.1.0"]
+       def runId = githubWorkflowRun.triggerGithubActionsWorkflow(args)
+       def runInfo = githubWorkflowRun.getWorkflowRun(args + [runId: runId]) 
+   }
+*/
+
+import groovy.json.JsonSlurperClassic 
+
+def call(Map args = [:]) {
+    def buildTimeLimit = args.get('buildTimeLimit', 1)
+    def runId = triggerGithubActionsWorkflow(args)
+    def startDate = new Date()
+    while(startDate.time  + buildTimeLimit*3600000 -  new Date().time > 0) {
```

Comment:
Can you explain what's the intention with this loop? A comment could help

Reply:
Added comment

Reply:
I wonder if 1 hour could be a bit too much, what do you think to customise the default hour to be customised with an argument?

Reply:
changed buildTimeLimit unit to minutes and default value to 30.